google/cloud/internal/external_account_token_source_aws.cc

```diff
@@ -95,6 +115,94 @@ StatusOr<ExternalAccountTokenSourceAwsInfo> ParseExternalAccountTokenSourceAws(
       /*imdsv2_session_token_url=*/*std::move(imdsv2)};
 }
 
+StatusOr<std::string> FetchMetadataToken(
+    ExternalAccountTokenSourceAwsInfo const& info,
+    HttpClientFactory const& client_factory, Options const& opts,
+    internal::ErrorContext const& /*ec*/) {
+  if (info.imdsv2_session_token_url.empty()) return std::string{};
+  auto client = client_factory(opts);
+  auto request =
+      rest_internal::RestRequest()
+          .SetPath(std::move(info.imdsv2_session_token_url))
+          .AddHeader(kMetadataTokenTtlHeader,
+                     std::to_string(kDefaultMetadataTokenTtl.count()));
```

Comment:
Nit: In order to avoid depending on the exact `duration` type of a named object in the places where we need to convert it to a particular variety or count, I would always suggest something like
```
  std::chrono::duration_cast<std::chrono::seconds>(kDefaultMetadataTokenTtl).count()
```
In this case a glorified no-op, but it makes what is going on clearer I reckon (the alternative being putting units in the name, which feels like going in the wrong direction).